tests/unit/utils.py

```diff
@@ -22,55 +22,111 @@
 )
 
 
+def get_output_type(calculation_type: str, sym: bool) -> str:
+    if calculation_type == "short_circuit":
+        if sym:
+            raise AssertionError(f"Unsupported validation case: calculation_type={calculation_type}, sym={sym}")
+        return "sc_output"
+
+    if sym:
+        return "sym_output"
+
+    return "asym_output"
+
+
+def get_test_case_paths(calculation_type: str, test_cases: Optional[List[str]] = None) -> Dict[str, Path]:
+    """get a list of all cases, directories in validation datasets"""
+    calculation_type_dir = DATA_PATH / calculation_type
+    test_case_paths = {
+        str(item.relative_to(DATA_PATH)).replace("\\", "/"): item
+        for item in calculation_type_dir.glob("**/")
+        if (item.is_dir() and (item / "params.json").is_file())
+    }
+    if test_cases is not None:
+        test_case_paths = {key: value for key, value in test_case_paths.items() if key in test_cases}
+
+    return test_case_paths
+
+
+def add_case(
+    case_name: str,
+    case_dir: Path,
+    params: dict,
+    calculation_type: str,
+    is_batch: bool,
+    calculation_method: str,
+    sym: bool,
+):
+    output_prefix = get_output_type(calculation_type=calculation_type, sym=sym)
+    batch_suffix = "_batch" if is_batch else ""
+
+    # only generate a case if sym or asym output exists
+    if (case_dir / f"{output_prefix}{batch_suffix}.json").exists():
+        # Build a recognizable case ID
+        case_id = case_name
+        case_id += "-sym" if sym else "-asym"
+        case_id += "-" + calculation_method
+        if is_batch:
+            case_id += "-batch"
+        pytest_param = [
+            case_id,
+            case_dir,
+            sym,
+            calculation_type,
+            calculation_method,
+            params["rtol"],
+            params["atol"],
+        ]
+        kwargs = {}
+        if "fail" in params:
+            kwargs["marks"] = pytest.mark.xfail(reason=params["fail"], raises=AssertionError)
+        elif calculation_type == "short_circuit":
+            kwargs["marks"] = (pytest.mark.xfail(reason="Short circuit is not yet supported"),)
```

Comment:
I commented out this line to test all other changes. Doing so results in the expected error: power_grid_model.errors.PowerGridError: CalculationType is not implemented for __int64 #2!

Example:

FAILED tests/unit/test_0Z_model_validation.py::test_batch_validation[short_circuit/single_phase_to_ground_a-asym-iec60909-batch] - power_grid_model.errors.PowerGridError: CalculationType is not implemented for __int64 #2!
FAILED tests/unit/test_0Z_model_validation.py::test_batch_validation[short_circuit/single_phase_to_ground_b-asym-iec60909-batch] - power_grid_model.errors.PowerGridError: CalculationType is not implemented for __int64 #2!
FAILED tests/unit/test_0Z_model_validation.py::test_batch_validation[short_circuit/single_phase_to_ground_c-asym-iec60909-batch] - power_grid_model.errors.PowerGridError: CalculationType is not implemented for __int64 #2!
FAILED tests/unit/test_0Z_model_validation.py::test_batch_validation[short_circuit/three_phase_abc-asym-iec60909-batch] - power_grid_model.errors.PowerGridError: CalculationType is not implemented for __int64 #2!
FAILED tests/unit/test_0Z_model_validation.py::test_batch_validation[short_circuit/two_phase_ab-asym-iec60909-batch] - power_grid_model.errors.PowerGridError: CalculationType is not implemented for __int64 #2!
FAILED tests/unit/test_0Z_model_validation.py::test_batch_validation[short_circuit/two_phase_ac-asym-iec60909-batch] - power_grid_model.errors.PowerGridError: CalculationType is not implemented for __int64 #2!
FAILED tests/unit/test_0Z_model_validation.py::test_batch_validation[short_circuit/two_phase_bc-asym-iec60909-batch] - power_grid_model.errors.PowerGridError: CalculationType is not implemented for __int64 #2!
FAILED tests/unit/test_0Z_model_validation.py::test_batch_validation[short_circuit/two_phase_to_ground_ab-asym-iec60909-batch] - power_grid_model.errors.PowerGridError: CalculationType is not implemented for __int64 #2!
FAILED tests/unit/test_0Z_model_validation.py::test_batch_validation[short_circuit/two_phase_to_ground_ac-asym-iec60909-batch] - power_grid_model.errors.PowerGridError: CalculationType is not implemented for __int64 #2!
FAILED tests/unit/test_0Z_model_validation.py::test_batch_validation[short_circuit/two_phase_to_ground_bc-asym-iec60909-batch] - power_grid_model.errors.PowerGridError: CalculationType is not implemented for __int64 #2!

Note: I did not implement validate_input_data or validate_batch_data. Only the prep work for the regular calculation validation tests are implemented here.